homeassistant/components/broadlink/remote.py

```diff
@@ -0,0 +1,418 @@
+    unique_id = f"remote_{hexlify(mac_addr).decode('utf-8')}"
+
+    if unique_id in hass.data.setdefault(DOMAIN, {}).setdefault(COMPONENT, []):
+        _LOGGER.error("Duplicate: %s", unique_id)
+        return
+    hass.data[DOMAIN][COMPONENT].append(unique_id)
+
+    api = broadlink.rm((host, DEFAULT_PORT), mac_addr, None)
+    api.timeout = timeout
+    remote = BroadlinkRemote(name, unique_id, api, hass.config.path)
+
+    connected = False
+    loaded = False
+    try:
+        connected, loaded = await asyncio.gather(
+            hass.async_add_executor_job(api.auth), remote.async_retrieve_data()
+        )
+    except socket.error:
+        pass
+    if not connected:
+        hass.data[DOMAIN][COMPONENT].remove(unique_id)
+        raise PlatformNotReady
+    if not loaded:
+        _LOGGER.error("Failed to set up %s", unique_id)
+        hass.data[DOMAIN][COMPONENT].remove(unique_id)
+        return
+    async_add_entities([remote], False)
+
+
+class BroadlinkRemote(RemoteDevice):
+    """Representation of a Broadlink remote."""
+
+    def __init__(self, name, unique_id, api, config_path):
+        """Initialize the remote."""
+        self._name = name
+        self._unique_id = unique_id
+        self._api = api
+        self._command_file = config_path(f"{unique_id}.json")
+        self._toggle_file = config_path(f"{unique_id}.dat")
+        self._commands = {}
+        self._toggle = {}
+        self._state = True
+        self._available = True
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the remote."""
+        return self._unique_id
+
+    @property
+    def is_on(self):
+        """Return True if the remote is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if the remote is available."""
+        return self._available
+
+    @property
+    def should_poll(self):
+        """Return True if the remote has to be polled for state."""
+        return True
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_LEARN_COMMAND
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the remote on."""
+        self._state = True
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the remote off."""
+        self._state = False
+
+    async def async_update(self):
+        """Update the availability of the remote."""
+        if not self.available:
+            await self._async_connect()
+
+    async def async_send_command(self, command, **kwargs):
+        """Send a list of commands to a device."""
+        kwargs[ATTR_COMMAND] = command
+        kwargs = MINIMUM_SERVICE_SCHEMA(kwargs)
+        commands = kwargs.get(ATTR_COMMAND)
+        device = kwargs.get(ATTR_DEVICE)
+        repeat = kwargs.get(ATTR_NUM_REPEATS, DEFAULT_NUM_REPEATS)
```

Comment:
Please put default values in the validation schema.

Reply:
Done.